### ISSUE TITLE
Benchmarks for GM segmentation

### ISSUE THREAD

**Issue by Nilser3** [ivadomed/model-gm-contrast-region-agnostic#3]
# Description

To segment the human in-vivo spinal cord GM, there are pre-trained models, the idea is to test those models on our database (according to contrast).

# Models 
1. `sct_deepseg_gm` -> T2star like contrasts (GM bright, WM dark)
2. `seg_gm_sc_7t_t2star` -> 7T T2Star 
3.

It seems to me that they developed other models with single contrasts ( `lumbar-vanderbilt` @NathanMolinier ,  `dcm-brno` @valosekj  ) , do you have those models available somewhere?

Thank you.

# Related issues
#2

**Comment by valosekj** [comment]
> dcm-brno @valosekj ) , do you have those models available somewhere?

@KaterinaKrejci231054 has developed a nnUNet model for GM segmentation from `dcm-brno` T1w images. @KaterinaKrejci231054, can you please get in touch with Nilser and share the model with him?

**Comment by NathanMolinier** [comment]
The two models developed by Laurine are available [here](https://github.com/ivadomed/lumbar-T2star-GMseg/releases/tag/r20230601)

**Comment by Nilser3** [comment]
# Model `seg_gm_psir` (benchmark for PSIR images )

- Based on `hc-ucsf-psir` dataset (train/valid on 88 subjects / 88 2D images)
- 2D nnUnet model 
- Trained on 250 epochs 
- Available [here](https://amubox.univ-amu.fr/s/yymK4WrqXm2XNDb)

## Test on a external dataset:

- Dataset `ms-barcelona-psir` (66 subjects / 66 2D images)
- Segmentation methods 
1. `seg_gm_psir`
2. `contrast-region-agnostic model`: Details [here](https://github.com/ivadomed/model-gm-contrast-region-agnostic/issues/2#issuecomment-2539496534) 
<img src="https://github.com/user-attachments/assets/f64194cf-c64a-4d56-99cf-e76a78b8b41c" alt="Image" width="400"/>

Both methods have comparable results, so I consider that we can continue with a `contrast-region-agnostic` segmentation approach.

**Comment by Nilser3** [comment]
> [@KaterinaKrejci231054](https://github.com/KaterinaKrejci231054) has developed a nnUNet model for GM segmentation from `dcm-brno` T1w images. 

# Benchmark for T1w images 

- Models developed by @KaterinaKrejci231054 
- Based on `dcm-brno` dataset
-- `seg_gm_t1w-single-class`: 3D nnUnet model, available [here](https://amubox.univ-amu.fr/s/mEKRpbmLHsjTJbQ)
-- `seg_gm_t1w-multi-class`: 3D nnUnet region-based model, available [here](https://amubox.univ-amu.fr/s/Kam8ctNis4ro3eN) 
- Trained on 500 epochs 
- I dont know which subjects were used for train/valid (possible bias when comparing with another method).

## Test on `dcm-brno` test split 
(see [Data-gm-contrast-agnostic_from_Table_01.xlsx](https://github.com/ivadomed/model-gm-contrast-region-agnostic/blob/nlm/first_model/codes/Data-gm-contrast-agnostic_from_Table_01.xlsx))
15 subjects / 629 2D images
- Segmentation methods:
1. `seg_gm_t1w-single-class`
2. `seg_gm_t1w-multi-class`
3. `contrast-region-agnostic model`: Details [here](https://github.com/ivadomed/model-gm-contrast-region-agnostic/issues/2#issuecomment-2539496534) 

<img src="https://github.com/user-attachments/assets/c089c714-a9e9-47af-ac42-549766fb6464" alt="Image" width="500"/>

There are many Dices Scores at 1 for `seg_gm_t1w` methods, possibly because we are testing on the train/valid dataset.
however the `contrast-region-agnostic` method has a distribution more centered.

**Comment by valosekj** [comment]
Nice results, @Nilser3!

> I dont know which subjects were used for train/valid (possible bias when comparing with another method).

> There are many Dices Scores at 1 for seg_gm_t1w methods, possibly because we are testing on the train/valid dataset.

Here you are talking about _**train/val**_. But you want to evaluate the models on _**test**_ split, right?
I believe @KaterinaKrejci231054 has details about train and test splits for her models (`seg_gm_t1w-single-class`, `seg_gm_t1w-multi-class`). Although she used different subject naming for her models (trained ~3y ago), I sent her a transcript table to match subject names between the dataset she used 3y ago and the current git-annex `dcm-brno`.

**Comment by Nilser3** [comment]
> Here you are talking about _**train/val**_. But you want to evaluate the models on _**test**_ split, right? 

Exactly, I would like to know which subjects were used for testing, in order to make an objective comparison, thank you!

**Comment by KaterinaKrejci231054** [comment]
I am sorry for the delay - I shared the split file with Nilser on Slack.

**Comment by jcohenadad** [comment]
> There are many Dices Scores at 1 for seg_gm_t1w methods, possibly because we are testing on the train/valid dataset.
however the contrast-region-agnostic method has a distribution more centered.

that sounds very likely indeed

**Comment by Nilser3** [comment]
#  Benchmark for T1w images

Thank you @KaterinaKrejci231054  for pointing out the subjects used for testing on `seg_gm_t1w` models, 

- Testing split for `seg_gm_t1w` models: `sub-2419B`, `sub-2420B`, `sub-2421B`, `sub-2451B`, `sub-2472B`, `sub-2649B`, `sub-2737B`, `sub-3066B`, `sub-3148B`, `sub-3194B`, `sub-3636B`

- Testing split for `contrast-region-agnostic` model: `sub-2271B`, `sub-2346B`, `sub-2414B`, `sub-2648B`, `sub-2698B`, `sub-2699B`, `sub-2756B`, `sub-2999B`, `sub-3094B`, `sub-3121B`, `sub-3170B`, `sub-3209B`, `sub-3640B`, `sub-3641B`

Indeed, we dont have any testing subject in common, which makes our comparison not objective.

**Comment by Nilser3** [comment]
# Benchmark for Lumbar T2star images

- Models developed by Laurine Thyssen
- Based on `lumbar-vanderbilt` dataset
-- `seg_gm_t2s-lumbar`: From their [repo](https://github.com/ivadomed/lumbar-T2star-GMseg), the `Dataset503` performed the `best` model with a 3D single class configuration. available: [r20230601](https://github.com/ivadomed/lumbar-T2star-GMseg/releases/tag/r20230601)
- Trained on 1000 epochs 
- I dont know which subjects were used for testing (possible bias when comparing with our method). @NathanMolinier is it possible to know on which subjects the `seg_gm_t2s-lumbar` model was tested?

## Test on `lumbar-vanderbilt` test split 
Subjects : `sub-140803`, `sub-141763`, `sub-241981`, `sub-242174`, `sub-242714`, `sub-243445`, `sub-243479`, `sub-245558`, `sub-245609`, `sub-246626`, `sub-247285`, `sub-247581`  (12 subjects / 166 2D images)
- Segmentation methods:
1. `seg_gm_t2s-lumbar`
2. `contrast-region-agnostic model`: Details [here](https://github.com/ivadomed/model-gm-contrast-region-agnostic/issues/2#issuecomment-2539496534) 

<img src="https://github.com/user-attachments/assets/fff93704-0043-4b0f-837d-72a856c5ebf9" alt="Image" width="400"/>

The `contrast-region-agnostic` model is expected to improve with the addition of synthesized lumbar data (see #5 ) in the training of a new version of this model.

**Comment by Nilser3** [comment]
# Model `seg_gm_7T-T1map` (benchmark for 7T-T1map images )

- Based on `marseille-7T-MP2RAGE` dataset 
- 2D nnUnet region-based model 
- Trained on 250 epochs 
- Available [here](https://amubox.univ-amu.fr/s/3a7jxH3TWm3Kcag)
- Same Testing subjects for `seg_gm_7T-T1map` and  `contrast-region-agnostic` models.

## Test on  `marseille-7T-MP2RAGE` testing dataset 
 (30 subjects / 368 2D images)
- Segmentation methods :
1. `seg_gm_7T-T1map`
2. `contrast-region-agnostic model`: Details [here](https://github.com/ivadomed/model-gm-contrast-region-agnostic/issues/2#issuecomment-2539496534) 

<img src="https://github.com/user-attachments/assets/5c76e9e6-a31d-44c0-816f-f374df0b61f5" alt="Image" width="400"/>

The benchmark seems to perform better, however, since for both models there are many Dices Scores at 0, the GTs will be reviewed. 

TODO: Review GM GTs

**Comment by jcohenadad** [comment]
@Nilser3 about https://github.com/ivadomed/model-gm-contrast-region-agnostic/issues/3#issuecomment-2577248404, how were your able to plot dice scores if you didn't know the test subjects for seg_gm_t1w? I thought you were using the same test subjects (that you selected).

**Comment by Nilser3** [comment]
> [@Nilser3](https://github.com/Nilser3) about [#3 (comment)](https://github.com/ivadomed/model-gm-contrast-region-agnostic/issues/3#issuecomment-2577248404), how were your able to plot dice scores if you didn't know the test subjects for seg_gm_t1w? I thought you were using the same test subjects (that you selected).

I apologize for not being clear @jcohenadad ,
I calculated dice scores in the test split of the `contras-agnostic model` design, and unfortunately all these subjects participated in the Training of the `seg_gm_t1w` model.

**Comment by Nilser3** [comment]
# Improving  the model `seg_gm_7T-T1map` (benchmark for 7T-T1map images )

Exploring why both models have many Dice Scores at 0, I have seen that there are empty GM GT slices at the beginning and end of the images:

<img src="https://github.com/user-attachments/assets/0f92182d-bc8f-4058-aee3-d5b4c7e37912" alt="Image" width="300"/>

So a cropping was done along the Z-axis of the anat images based on the GT, and the becnhmark was retrained with those images:

- Cropped images from `marseille-7T-MP2RAGE` dataset 
- 2D nnUnet region-based model 
- Trained on 250 epochs 
- Same Testing subjects 

## Test on  `marseille-7T-MP2RAGE` testing dataset 
 (30 subjects / 368 2D images)
- Segmentation methods :
1. Improved `seg_gm_7T-T1map`
2. `contrast-region-agnostic model`: Details [here](https://github.com/ivadomed/model-gm-contrast-region-agnostic/issues/2#issuecomment-2539496534) , trained on ALL FULL FOV images 

<img src="https://github.com/user-attachments/assets/99c0a290-8aee-4c80-b2a4-e7a9b5eb6ca5" alt="Image" width="400"/>

The number of DICES at 0 has decreased significantly.
Next step: Re-training the `contrast-region-agnostic` model with cropped images on the GTs.
